aiperf/exporters/gpu_telemetry_console_exporter.py

```diff
@@ -0,0 +1,232 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+from rich.console import Console, Group, RenderableType
+from rich.table import Table
+from rich.text import Text
+
+from aiperf.common.decorators import implements_protocol
+from aiperf.common.enums.data_exporter_enums import ConsoleExporterType
+from aiperf.common.factories import ConsoleExporterFactory
+from aiperf.common.mixins import AIPerfLoggerMixin
+from aiperf.common.models.telemetry_models import TelemetryResults
+from aiperf.common.protocols import ConsoleExporterProtocol
+from aiperf.exporters.display_units_utils import normalize_endpoint_display
+from aiperf.exporters.exporter_config import ExporterConfig
+
+
+@implements_protocol(ConsoleExporterProtocol)
+@ConsoleExporterFactory.register(ConsoleExporterType.TELEMETRY)
+class GPUTelemetryConsoleExporter(AIPerfLoggerMixin):
+    """Console exporter for GPU telemetry data.
+
+    Displays GPU metrics in a table format similar to other console exporters.
+    Only displays when verbose mode is enabled.
+    """
+
+    def __init__(self, exporter_config: ExporterConfig, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self._results = exporter_config.results
+        self._service_config = exporter_config.service_config
+        self._exporter_config = exporter_config
+        self._telemetry_results = exporter_config.telemetry_results
+
+    async def export(self, console: Console) -> None:
+        """Export telemetry data to console if verbose mode is enabled.
+
+        Only displays telemetry data when verbose mode is enabled in service config.
+        Skips display if no telemetry data is available.
+
+        Args:
+            console: Rich Console instance for formatted output
+        """
+
+        if not self._service_config.verbose:
+            return
+
+        if not self._telemetry_results:
+            return
+
+        self._print_renderable(console, self.get_renderable(self._telemetry_results))
+
+    def _print_renderable(self, console: Console, renderable: RenderableType) -> None:
+        """Print the renderable to the console with formatting.
+
+        Adds blank line before output and flushes console buffer after printing.
+
+        Args:
+            console: Rich Console instance for formatted output
+            renderable: Rich renderable object (Table, Group, Text, etc.) to display
+        """
+        console.print("\n")
+        console.print(renderable)
+        console.file.flush()
+
+    def get_renderable(self, telemetry_results: TelemetryResults) -> RenderableType:
+        """Create Rich tables showing GPU telemetry metrics with consolidated single-table format.
```

Comment:
this function is quite large. can you seperate it out into some sub functions that this calls?